### PR TITLE
Unionise Value

### PR DESCRIPTION
**Pull request by Hendiadyoin1**
This decreases the size of Value by unionising the Object* and the fast_integer.
The test-performance does not seem to be affected significantly by this, but it
is still nice to decrease the memory footprint a bit.

### CODE REVIEW

**Comment by seven1m** [comment]
Thanks!

**Comment by seven1m** [comment]
:thinking:

File: src/value.cpp [hunk 1]
```diff
@@ -29,7 +29,6 @@ void Value::hydrate() {
     case Type::Integer: {
         m_type = Type::Pointer;
         m_object = new IntegerObject { m_integer };
```
Comment (TheGrizzlyDev) [inline src/value.cpp:31]: c++ noob here, but isn't this implying that Value has both m_object and m_integer at the same time? Could it be that when building not in release that is possible and somehow does not complain but when running the release build the union gets compacted and thus you'd end up with some kind of undefined behaviour?

Comment (seven1m) [reply src/value.cpp:31]: It's a good thought. We did try different ordering here, a temporary variable, etc., but it didn't seem to matter. But we may have missed something.

**Comment by seven1m** [comment]
This made a wee-bit of a speed up on the boardslam example code, so that's neat.

Screenshot from 2021-12-21 20-31-26

**Comment by seven1m** [comment]
Here's a slightly better speed-up from the [pemutation_spec.rb](https://github.com/seven1m/natalie/blob/master/spec/core/array/permutation_spec.rb):

Screenshot from 2021-12-21 21-10-35

**Pull request merged**